executor/join.go

```diff
@@ -185,10 +185,26 @@ func (e *HashJoinExec) getJoinKeyFromChkRow(isOuterKey bool, row chunk.Row, keyB
 	}
 
 	keyBuf = keyBuf[:0]
-	keyBuf, err = codec.HashChunkRow(e.ctx.GetSessionVars().StmtCtx, keyBuf, row, allTypes, keyColIdx)
-	if err != nil {
-		err = errors.Trace(err)
+	sc := e.ctx.GetSessionVars().StmtCtx
+	for idx, i := range keyColIdx {
+		d := row.GetDatum(i, allTypes[i])
```

Comment:
It's inefficient to convert the data inside a `Chunk` to `Datum` then convert the data to the target type. Can we directly convert the data inside the `Chunk` to the target type?